### PR TITLE
Feature: Separate fetching and waiting times for response

### PR DESCRIPTION
**Pull request by ringus1**
Adding just a tiny bit more granularity for metadata returned when executing requests. Similarly as browsers do this and provide info when server first responded and how long it took to actually fetch the content.

image

### CODE REVIEW

**Comment by cyberw** [comment]
I like this as a feature, but it needs more work + tests. also needs to be backwards compatible, which I dont think this is?

**Comment by ringus1** [comment]
I mean, sure it's not a fully ready "feature" per se, as it should be integrated with GUI, etc. But it is backwards compatible definately. It just adds additional kwargs to `request_metadata`, that's it. Can be used by people manually. And used in future PRs to extend UI.

**Comment by ringus1** [comment]
The only "incompatible" change would be removing cast to `int` of _response_time_ in `FastHttpSession`. But standard `HttpSession` returns `float` normally. So just wanted to make it standarized.

**Comment by cyberw** [comment]
I see. I misread the diff to imply that response_time was only set in certain conditions, sorry. 

I'm not sure about the naming

I wont have time to review this any time soon, but if you add unit tests I can check it out later.

**Comment by ringus1** [comment]
Ok, will do. Thank you.

File: locust/clients.py [hunk 1]
```diff
@@ -130,10 +130,16 @@ def request(self, method, url, name=None, catch_response=False, context={}, **kw
         # prepend url with hostname unless it's already an absolute URL
         url = self._build_url(url)
 
+        # use stream=True to avoid closing socket and downloading content at the beginning
+        # we'll download content if actually requested a bit later
+        #
+        # a bit of a hack to actually estimate fetching time
+        stream = kwargs.pop("stream", False)
```
Comment (ringus1) [inline locust/clients.py:137]: @cyberw I really hate this one, but it seems standard `requests.Session` fetches the content eagerly when `stream=False` is used. So the only way to actually estimate `fetching` is to use `stream=True` during first request. Then consume `content` normally later.

Comment (cyberw) [reply locust/clients.py:137]: Does setting stream=True affect the headers sent, as this implies?https://stackoverflow.com/questions/60343944/how-does-requests-stream-true-option-streams-data-one-block-at-a-time
If it does then I dont think this change is ok...

**Pull request closed without merging**